From: Orit Wasserman <owasserm@redhat.com>
Date: Wed, 19 Oct 2016 21:51:01 +0200
Subject: rgw: fix put_acls for objects starting and ending with underscore

Fixes: http://tracker.ceph.com/issues/17625
Signed-off-by: Orit Wasserman <owasserm@redhat.com>
(cherry picked from commit 14d4d912c7b47c56b16ae0bdc6bc08d208de3461)

Resolves: rhbz#1372346
---
 src/rgw/rgw_op.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/rgw/rgw_op.cc b/src/rgw/rgw_op.cc
index 31a143b..bc1e75a 100644
--- a/src/rgw/rgw_op.cc
+++ b/src/rgw/rgw_op.cc
@@ -3556,7 +3556,8 @@ void RGWPutACLs::execute()
   }
 
   new_policy.encode(bl);
-  obj = rgw_obj(s->bucket, s->object);
+  obj = rgw_obj(s->bucket, s->object.name);
+  obj.set_instance(s->object.instance);
   map<string, bufferlist> attrs;
 
   store->set_atomic(s->obj_ctx, obj);
